src/Razor/test/Microsoft.VisualStudio.Razor.Integration.Test/Semantic/RazorSemanticTokensTests.cs

```diff
@@ -0,0 +1,183 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT license. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Reflection;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Razor.Language;
+using Microsoft.VisualStudio.Razor.Integration.Test.InProcess;
+using Microsoft.VisualStudio.Text;
+using Microsoft.VisualStudio.Text.Classification;
+using Xunit;
+
+namespace Microsoft.VisualStudio.Razor.Integration.Test
+{
+    [IntializeTestFile]
+    public class RazorSemanticTokensTests : AbstractRazorEditorTest
+    {
+        private static readonly AsyncLocal<string?> s_fileName = new();
+
+        private static readonly string s_projectPath = TestProject.GetProjectDirectory(typeof(RazorSemanticTokensTests), useCurrentDirectory: true);
+
+        protected bool GenerateBaselines { get; set; } = false;
+
+        // Used by the test framework to set the 'base' name for test files.
+        public static string? FileName
+        {
+            get { return s_fileName.Value; }
+            set { s_fileName.Value = value; }
+        }
+
+        public override async Task InitializeAsync()
+        {
+            await base.InitializeAsync();
+            await TestServices.Workspace.WaitForAsyncOperationsAsync(FeatureAttribute.Classification, HangMitigatingCancellationToken);
+        }
+
+        [IdeFact]
+        public async Task Components_AreColored()
+        {
+            // Arrange
+            await TestServices.SolutionExplorer.OpenFileAsync(BlazorProjectName, MainLayoutFile, HangMitigatingCancellationToken);
+
+            // Act
+            await TestServices.Editor.WaitForClassificationAsync(HangMitigatingCancellationToken, "RazorComponentElement");
+
+            // Assert
+            var expectedClassifications = await GetExpectedClassificationSpansAsync(nameof(Components_AreColored), HangMitigatingCancellationToken);
+            await TestServices.Editor.VerifyGetClassificationsAsync(expectedClassifications, HangMitigatingCancellationToken);
```

Comment:
👏 

If for some reason each of these tests take too long on their own I also wouldn't object to having a super "things are colored correctly" test

Reply:
The individual tests are actually really snappy on my local machine, it's the VS initialization that takes a minute, but thankfully that only needs to happen once per run.